index.js

```diff
@@ -11,11 +11,21 @@ function toObject(val) {
 	return Object(val);
 }
 
-function isFunc(val) {
-	return typeof val === 'function';
+function assignKey(to, from, key) {
+	var val = from[key];
+
+	if (typeof val === 'undefined' || val === null) {
+		return;
+	}
+
+	if (typeof to[key] === 'undefined' || !isObj(val)) {
```

Comment:
@schnittstabil Yes, but I think it would make deepAssign useless. Maybe we should throw an error in case of `null` but not of `undefined`. Not sure.

That would be really great to know what's the original purpose of deepAssign according to @sindresorhus. It's not good for deep clone an object but if you want to follow the specs it becomes useless (imho). 

Maybe, that would be the best if we didn't throw errors in case of `null` and `undefined`.